client/src/gameLogic.tsx

```diff
@@ -86,37 +144,43 @@ export function setPlayerWord(room: StartingPhase, playerName: string, word: str
 
 // Move a room from the StartingPhase to the HintingPhase.
 export function startGameRoom(room: StartingPhase): HintingPhase {
-    return {
+    let players = room.players.map((player, index) => {
+        // each player receives the previous player's word
+        const word = room.players[(index + room.players.length - 1) % room.players.length].word;
+        if (word == null) {
+            throw new Error("Room is not ready");
+        }
+        return {
+            name: player.name,
+            hand: {
+                letters: shuffle(word),
+                guesses: Array.from(word, _ => null),
+                activeIndex: 0,
+            },
+            hintsGiven: 0,
+        };
+    });
+
+    let deck = createShuffledDeck(flatten(players.map((player) => player.hand.letters)));
+    let dummies = dummyLettersForFreeHint(players.length).map((untilFreeHint) => ({
+        currentLetter: '#',    // placeholder to be replaced by cycleDummies
+        untilFreeHint: untilFreeHint + 1,
+    }));
```

Comment:
this is jank

Reply:
rip i wanted to reuse the draw dummies logic which is itself big jank